tests/test_other.py

```diff
@@ -7914,77 +7914,89 @@ def args_to_filename(args):
     if expected_funcs is not None:
       self.assertEqual(funcs, expected_funcs)
 
-  def test_binaryen_metadce_minimal(self):
-    def run(*args):
-      self.run_metadce_test('minimal.c', *args)
-
-    if self.is_wasm_backend():
-      run([],      11, [], ['waka'],  9211,  5, 13, 16) # noqa
-      run(['-O1'],  9, [], ['waka'],  7886,  2, 12, 10) # noqa
-      run(['-O2'],  9, [], ['waka'],  7871,  2, 12, 10) # noqa
-      # in -O3, -Os and -Oz we metadce, and they shrink it down to the minimal output we want
-      run(['-O3'],  0, [], [],          85,  0,  2,  2) # noqa
-      run(['-Os'],  0, [], [],          85,  0,  2,  2) # noqa
-      run(['-Oz'],  0, [], [],          54,  0,  1,  1) # noqa
-    else:
-      run([],      21, ['abort'], ['waka'], 22712, 22, 15, 30) # noqa
-      run(['-O1'], 10, ['abort'], ['waka'], 10450,  7, 11, 11) # noqa
-      run(['-O2'], 10, ['abort'], ['waka'], 10440,  7, 11, 11) # noqa
-      # in -O3, -Os and -Oz we metadce, and they shrink it down to the minimal output we want
-      run(['-O3'],  0, [],        [],          55,  0,  1, 1) # noqa
-      run(['-Os'],  0, [],        [],          55,  0,  1, 1) # noqa
-      run(['-Oz'],  0, [],        [],          55,  0,  1, 1) # noqa
+  @parameterized({
+    'O0': ([],      11, [], ['waka'],  9211,  5, 13, 16), # noqa
+    'O1': (['-O1'],  9, [], ['waka'],  7886,  2, 12, 10), # noqa
+    'O2': (['-O2'],  9, [], ['waka'],  7871,  2, 12, 10), # noqa
+    # in -O3, -Os and -Oz we metadce, and they shrink it down to the minimal output we want
+    'O3': (['-O3'],  0, [], [],          85,  0,  2,  2), # noqa
+    'Os': (['-Os'],  0, [], [],          85,  0,  2,  2), # noqa
+    'Oz': (['-Oz'],  0, [], [],          54,  0,  1,  1), # noqa
+  })
+  @no_fastcomp()
+  def test_binaryen_metadce_minimal(self, args, sent, exists, not_exists, size, imports, exports, funcs):
+    self.run_metadce_test('minimal.c', args, sent, exists, not_exists, size, imports, exports, funcs)
```

Comment:
To me this is more of a "a struct would be useful here" type of code smell. I have a mild preference for letting it be this ugly, because it actually is this ugly.

From  http://www.cs.yale.edu/homes/perlis-alan/quotes.html:

> 11. If you have a procedure with ten parameters, you probably missed some.